File: src/send.cc
#include <cstdio>
#include <gflags/gflags.h>

#include "ext/cpr.h"
#include "ext/json.h"
#include "ext/sprintf.h"

DEFINE_string(server_host, "localhost", "Server host");
DEFINE_int32(server_port, 1337, "Server port");

int main(int argc, char* argv[]) {
    gflags::ParseCommandLineFlags(&argc, &argv, true);
    json x = {
        {"a", 123.456},
        {"b", json{1, 2, 3, "Go!"}},
    };
    auto to = StringPrintf("http://%s:%d/json", FLAGS_server_host.c_str(),
                           FLAGS_server_port);
    auto ret = cpr::Post(cpr::Url(to), cpr::Body{x.dump()},
                         cpr::Header{{"Content-Type", "text/plain"}});
    printf("Status code: %d\n", ret.status_code);
    printf("Body: %s\n", ret.text.c_str());
}

File: py/send.py
from argparse import ArgumentParser
import requests
import ujson as json


def parse_args():
    ap = ArgumentParser()
    ap.add_argument('--server_host', type=str, default='localhost',
                    help='Server host.')
    ap.add_argument('--server_port', type=int, default=1337,    
                    help='Server port.')
    return ap.parse_args()


def run(args):
    x = {
        'a': 123.456,
        'b': [1, 2, 3, 'Go!'],
    }
    url = 'http://%s:%d/json' % (args.server_host, args.server_port)
    ret = requests.post(url, json.dumps(x))
    print('Status code: %d' % ret.status_code)
    print('Body: %s' % ret.text)


if __name__ == '__main__':
    run(parse_args())
